bool finish_training
---
float32[] state